### PR TITLE
Add Product details

### PR DESCRIPTION
**Pull request by alouanemed**

### CODE REVIEW

File: OnlineStoreTCA/Products/ProductList/ProductListView.swift [hunk 1]
```diff
@@ -32,9 +32,13 @@ struct ProductListView: View {
                                     action: ProductListDomain.Action
                                         .product(id: action:)
                                 )
-                            ) {
-                                ProductCell(store: $0)
-                            }
+                            ) { cellStore in
+                                NavigationLink {
+                                    ProductDetailsView(store: cellStore)
```
Comment (alouanemed) [inline OnlineStoreTCA/Products/ProductList/ProductListView.swift]: @pitt500 here^

**Comment by alouanemed** [comment]
@pitt500 
I am somewhat blocked on which way to provide the Product Details Store in the product list. Since you proposed to have a separate store for the details, otherwise we could use the ProductDomain.
So how to provide the store in question ?

**Comment by pitt500** [comment]
First off, thank you very much for your time on this @alouanemed! And about you question, What is the problem that block you to do the Detail Domain?

File: OnlineStoreTCA/Products/ProductDetails/ProductDetailsViewDomain.swift [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+//  ProductDetailsViewDomain.swift
+//  OnlineStoreTCA
+//
+//  Created by Mohamed Alouane on 9/10/2022.
+//
+
+import Foundation
+import ComposableArchitecture
+
+struct ProductDetailsDomain{
```
Comment (pitt500) [inline OnlineStoreTCA/Products/ProductDetails/ProductDetailsViewDomain.swift:11]: Let's keep this singular

File: OnlineStoreTCA/Products/ProductDetails/ProductDetailsViewDomain.swift [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        var product: Product
+        var dataLoadingStatus = DataLoadingStatus.notStarted
+       
+        var isLoading: Bool {
+            dataLoadingStatus == .loading
+        }
+    }
+    
+    enum Action: Equatable{
+        case fetchProduct
```
Comment (pitt500) [inline OnlineStoreTCA/Products/ProductDetails/ProductDetailsViewDomain.swift:22]: I think we don't need to fetch the product again for this demo. Let's keep it simple for now and just pass the data in memory from the selected product.

File: OnlineStoreTCA/Products/ProductList/ProductListDomain.swift [hunk 4]
```diff
@@ -13,6 +13,7 @@ struct ProductListDomain {
         var dataLoadingStatus = DataLoadingStatus.notStarted
         var shouldOpenCart = false
         var cartState: CartListDomain.State?
+        var detailsState: ProductDetailsDomain.State?
```
Comment (pitt500) [inline OnlineStoreTCA/Products/ProductList/ProductListDomain.swift:16]: Singular too.

File: OnlineStoreTCA/Products/ProductList/ProductListView.swift (hunk 1 above)
Comment (pitt500) [reply OnlineStoreTCA/Products/ProductList/ProductListView.swift]: What is the problem here?

Comment (alouanemed) [reply OnlineStoreTCA/Products/ProductList/ProductListView.swift]: the `cellStore` here is type of `ProductDomain` while we want to use `ProductDetailsViewDomain`.

Comment (pitt500) [reply OnlineStoreTCA/Products/ProductList/ProductListView.swift]: Let me take a look

**Comment by pitt500** [comment]
Hey @alouanemed , sorry I'm in a coming release so my bandwidth is limited. I will take a look to this by the Weekend.

**Comment by pitt500** [comment]
Hi @alouanemed, I was able to fix the compilation issue and navigation now works, however, the product is not visible yet.

Take a look to this case study: https://github.com/pointfreeco/swift-composable-architecture/blob/main/Examples/CaseStudies/SwiftUICaseStudies/03-Navigation-Lists-NavigateAndLoad.swift

I used it to fix the issue, but I still need to refactor to get the right product (and ID). Let me know if you still any issues.

By the way, please rebase because I pushed some changes to the main repo yesterday.

**Comment by pitt500** [comment]
@alouanemed Do you have any question?

**Comment by pitt500** [comment]
@alouanemed Any updates yet?

**Comment by pitt500** [comment]
Hi @alouanemed, Since I haven't received any updates from this PR in a while, I will proceed to close it. Let me know if you want to reopen it. Thanks!

**Pull request closed without merging**